cpp/src/parquet/file_reader.cc

```diff
@@ -400,6 +401,21 @@ class SerializedFile : public ParquetFileReader::Contents {
     PARQUET_THROW_NOT_OK(cached_source_->Cache(ranges));
   }
 
+  ::arrow::Result<std::vector<::arrow::io::ReadRange>> GetReadRanges(
+      const std::vector<int>& row_groups, const std::vector<int>& column_indices,
+      int64_t hole_size_limit, int64_t range_size_limit) {
+    std::vector<::arrow::io::ReadRange> ranges;    
+    for (int row : row_groups) {
```

Comment:
`row_group` may be better than `row` because "row" and "row_group" are different in the Parquet context. 

```suggestion
    for (int row_group : row_groups) {
```